unstable/lens/struct-struct.rkt

```diff
@@ -0,0 +1,45 @@
+#lang sweet-exp racket/base
+
+provide struct->struct-lens
+
+require racket/local
+        unstable/lens/isomorphism/base
+        for-syntax racket/base
+                   syntax/parse
+                   "private/struct-id.rkt"
+module+ test
+  require lens/private/base/base
+          rackunit
+
+(define-syntax struct->struct-lens
+  (lambda (stx)
+    (syntax-parse stx
+      [(struct->struct-lens s1:struct-id s2:struct-id)
+       (unless (= (length (syntax->list #'[s1.accessor-id ...]))
+                  (length (syntax->list #'[s2.accessor-id ...])))
+         (raise-syntax-error #f (format "~a and ~a have different numbers of fields"
+                                        (syntax-e #'s1) (syntax-e #'s2))
+                             stx))
+       #'(local [(define (s1->s2 struct1)
+                   (#%app s2.constructor-id (s1.accessor-id struct1) ...))
```

Comment:
I'm not sure, but I suspect it's because the constructor-id from the struct-info ends up pointing to an identifier macro or something like that that messes it up in some way. I forget.